parts/linux/cloud-init/artifacts/aks-check-network.sh

```diff
@@ -0,0 +1,206 @@
+#! /bin/bash
+# 
+# AKS Check Network
+#
+# This script is used to check network connectivity from the node to certain required AKS endpoints 
+# and log the results to the events directory. For now, this script has to be triggered manually to
+# collect the log. In the future, we will run it periodically to check and alert any issue.
+
+APISERVER_FQDN=${1:-''}
+CUSTOM_ENDPOINT=${2:-''}
+
+EVENTS_LOGGING_PATH="/var/log/azure/Microsoft.Azure.Extensions.CustomScript/events/"
+AZURE_CONFIG_PATH="/etc/kubernetes/azure.json"
+AKS_CA_CERT_PATH="/etc/kubernetes/certs/apiserver.crt"
+AKS_CERT_PATH="/etc/kubernetes/certs/client.crt"
+AKS_KEY_PATH="/etc/kubernetes/certs/client.key"
+RESOLV_CONFIG_PATH="/etc/resolv.conf"
+SYSTEMD_RESOLV_CONFIG_PATH="/run/systemd/resolve/resolv.conf"
+
+ARM_ENDPOINT="management.azure.com"
+METADATA_ENDPOINT="http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=https://${ARM_ENDPOINT}/"
+API_VERSION="2023-11-01"
```

Comment:
nit: are we going to periodically update this API version? if not, we can just hardcode like its done for metadata_endpoint, otherwise, we should think of a mechanism to update this through some automation. can be a follow up, not blocking for this PR.

Reply:
we might but that won't be necessary for now so I removed `API_VERSION` variable according to your suggestion.